src/python/pants/base/exception_sink.py

```diff
@@ -354,7 +354,7 @@ def handle_signal_gracefully(cls, signum, frame):
     """
     if signum in cls.KEYBOARD_INTERRUPT_SIGNALS:
       raise KeyboardInterrupt('User interrupted execution with control-c!')
-    tb = frame.f_exc_traceback or sys.exc_info()[2]
+    tb = sys.exc_info()[2]
```

Comment:
What's the motivation for removing this? No need for a comment since it's deleted code, but I'm curious why it's necessary.

Also maybe add the rationale to your PR description.

Reply:
Ah yes! Thanks for reminding me. I put it commit, but forgot to put in PR 🤦‍♂️

Reply:
I'll add a little more context here. 
As far as I can tell, `f_exc_traceback` is undocumented (in both py2 and py3). It also seems to have been removed in python 3. 

I originally wanted to replace it with something similar that would exist in python 3. However, as I could not find it in docs and when I tried looking at it's content in and interpreter[1], it was empty. I decided to just remove it 

[1] Code sample (Python 2.7.10):
```python
In [1]: try:
    ...:     raise Exception('Yo')
    ...: except Exception:
    ...:     f = sys._getframe()
    ...:     print(f.f_exc_traceback)
    ...:     print(sys.exc_info()[2])
    ...: 
None
<traceback object at 0x102a29518>
```

Reply:
Same code in python3 causes this exception:
```python3
<ipython-input-30-9c028954dcfb> in <module>
      3 except Exception:
      4      f = sys._getframe()
----> 5      print(f.f_exc_traceback)
      6      print(sys.exc_info()[2])

AttributeError: 'frame' object has no attribute 'f_exc_traceback'
```